pkg/state/state_ofed.go

```diff
@@ -456,12 +460,22 @@ func (s *stateOFED) getMofedDriverImageName(cr *mellanoxv1alpha1.NicClusterPolic
 	}
 	reqLogger.V(consts.LogLevelDebug).Info("Generating ofed driver image name for version: %v", "version", curDriverVer)
 
+	oldRhelFormatDriverVer, err := semver.NewVersion(oldMofedRhelVersion)
+	if err != nil {
+		reqLogger.V(consts.LogLevelDebug).Info("failed to parse ofed driver version as semver")
+	}
+	osName := nodeAttr[nodeinfo.AttrTypeOSName]
+	osVer := nodeAttr[nodeinfo.AttrTypeOSVer]
+	if curDriverVer != nil && curDriverVer.GreaterThan(oldRhelFormatDriverVer) && osName == rhelOSName {
+		osVer = nodeAttr[nodeinfo.AttrTypeOSVerMajor] + ".x"
```

Comment:
I asked Vlad to tag (re-tag) each version explicitly why do we need this.

in addition IF we will need a special version for a specific RHEL release we will loose the ability to do it.

Reply:
@e0ne i think we should try to keep a single format in network operator.

MOFED driver container can be re-tagged as needed for multiple OS versions